src/Haskell/Ide/Engine/Dispatcher.hs

```diff
@@ -49,9 +51,30 @@ ideDispatcher env pin = forever $ do
   (IdeRequest lid callback action) <- liftIO $ atomically $ readTChan pin
   debugm $ "ideDispatcher:got request with id: " ++ show lid
   cancelled <- liftIO $ atomically $ isCancelled env lid
-  unless cancelled $ do
-    res <- action
-    liftIO $ callback res
+  res <- action
```

Comment:
This looks wrong- we definitely should not be running actions that have been cancelled.